articles/building-an-appointment-scheduling-telegram-bot/index.md

```diff
@@ -0,0 +1,514 @@
+
+Next, you updated the `last_command` field in the user's data to save the `add_appointment` command the user sent. You then sent a message requesting the user to enter the name of the appointment and its due date separated by a comma.
+
+You have to create another method called `echo` that will collect the name and the due date for the new appointment then save it to the database. Copy and paste the code below after your `add_appointment` method in your python file.
+
+
+```python
+
+def echo(update, context):
+
+    chat_id = update.effective_chat.id
+
+    message = update.message.text
+
+
+    user = client.query(q.get(q.match(q.index("users_index"), chat_id)))
+
+    last_command = user["data"]["last_command"]
+
+
+    if last_command == "add_appointment":
+
+        events = client.query(q.create(q.collection("Appointments"), {
+
+            "data": {
+
+                "user_id": chat_id,
+
+                "event": message.split(",")[0],
+
+                "completed": False,
+
+                "date_due": message.split(",")[1]
+
+            }
+
+        }))
+
+        client.query(q.update(q.ref(q.collection("Users"), user["ref"].id()), {"data": {"last_command": ""}}))
+
+        context.bot.send_message(chat_id=chat_id, text="Successfully added appointment event 👍")
+
+```
+
+In the code above, you defined a method called `echo` where you passed `update` and `context` as parameters. In the function, you saved the user’s message and `chat_id`. You then made use of the `chat_id` to query the FQL client using the `users_index` index to check the user data that matches the `chat_id`.
+
+Next, you checked if the `last_command` in the data retrieved is an `add_appointment` command. If it is, you proceed to query the FQL client to create an appointment using the Fauna `create` method in the `Appointments` collection with the information provided in the user’s message.
+
+Finally, you need to create a command handler for the `add_appointment` command and a message handler for the `echo` method. The message handler filters the message and triggers the `echo` method when a message the user enters a message( a message is any text that does not start with `/`).
+
+> Note: The echo method only works when the last command was an `add_appointment` command.
+
+```python
+
+dispatcher.add_handler(CommandHandler("add_appointment", add_appointment))
+
+dispatcher.add_handler(MessageHandler(Filters.text, echo))
+
+```
+
+![add_appointment](/engineering-education/building-an-appointment-scheduling-telegram-bot/add_appointment.png)
+
+#### Listing Appointments
+
+To list all the appointments saved for a user, copy and paste the code below in your python file.
+
```

Comment:
The following part will provide the feature of listing all the appointments saved for a user.